### DISCUSSION TITLE
What's the recommended way to work around IDE0058 for mocks?

### DISCUSSION THREAD

**Discussion by Bartleby2718** [devlooped/moq#1486] [Q&A] [answered]
If I do
```cs
mock.Setup(library => library.DownloadExists("2.0.0.0")).Returns(true);
```
with [`IDE0058`](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0058)'s severity set to error, you get an error with the message
> Expression value is never used

If I want to keep `IDE0058`, what do you think I should do here--just discard it (like `_ = mock.Setup(library => library.DownloadExists("2.0.0.0")).Returns(true);`)?

**Comment by kzu** [comment, marked as answer]
Yep, since you can keep invoking more methods after Returns, you'll get that warning. I'd disable it globally for all test projects, something like:

```editorconfig
# Test settings
[**/*Tests*/**{.cs,.vb}]
# Remove unnecessary expression value
dotnet_diagnostic.IDE0058.severity = none
```
